flyteplugins/go/tasks/pluginmachinery/tasklog/template.go

```diff
@@ -206,21 +235,44 @@
 		})
 	}
 
+	genericDynamicLogLinks := taskTemplateHasGenericDynamicLogLinksEnabled(input.TaskTemplate)
 	for _, dynamicLogLinkType := range getDynamicLogLinkTypes(input.TaskTemplate) {
 		for _, dynamicTemplateURI := range p.DynamicTemplateURIs {
-			if p.Name == dynamicLogLinkType {
-				for _, match := range taskConfigVarRegex.FindAllStringSubmatch(dynamicTemplateURI, -1) {
-					if len(match) > 1 {
-						if value, found := input.TaskTemplate.GetConfig()[match[1]]; found {
-							templateVars = append(templateVars, TemplateVar{MustCreateDynamicLogRegex(match[1]), value})
+			// In case there's a generic dynamic log link we should go over the configured dynamic log links
+			// in the task template and generate a link for each of them.
+			if p.Name == "Generic" {
+				for _, genericDynamicLogLink := range genericDynamicLogLinks {
+					if dynamicLogLinkType != genericDynamicLogLink.Name {
+						continue
+					}
+					for _, match := range taskConfigVarRegex.FindAllStringSubmatch(genericDynamicLogLink.TemplateURI, -1) {
+						if len(match) > 1 {
+							if value, found := input.TaskTemplate.GetConfig()[match[1]]; found {
+								templateVars = append(templateVars, TemplateVar{MustCreateDynamicLogRegex(match[1]), value})
+							}
+						}
+					}
+					taskLogs = append(taskLogs, &core.TaskLog{
+						Uri:           replaceAll(genericDynamicLogLink.TemplateURI, templateVars),
+						Name:          genericDynamicLogLink.DisplayName,
+						MessageFormat: core.TaskLog_JSON,
+					})
+				}
+			} else if p.Name == dynamicLogLinkType {
+				if p.Name == dynamicLogLinkType {
+					for _, match := range taskConfigVarRegex.FindAllStringSubmatch(dynamicTemplateURI, -1) {
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Remove redundant nested if condition</b></div>
<div id="fix">

Consider removing the redundant nested `if` condition since it's checking the same condition as the outer `if` statement
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
 			} else if p.Name == dynamicLogLinkType {
 					for _, match := range taskConfigVarRegex.FindAllStringSubmatch(dynamicTemplateURI, -1) {
				for _, match := range taskConfigVarRegex.FindAllStringSubmatch(dynamicTemplateURI, -1) {
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/flyteorg/flyte/pull/5066#issuecomment-2563965363>#51871a</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged